cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/validation/ValidationException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.etl.api.validation;
+
+import io.cdap.cdap.api.annotation.Beta;
+
+import java.util.List;
+
+/**
+ * Validation exception that carries multiple validation failures.
+ */
+@Beta
+public class ValidationException extends RuntimeException {
+  private List<ValidationFailure> failures;
+
+  /**
+   * Creates a validation exception with list of failures.
+   *
+   * @param failures list of validation failures
+   */
+  public ValidationException(List<ValidationFailure> failures) {
+    super(failures.isEmpty() ? "Validation Exception occurred." : failures.iterator().next().getMessage());
```

Comment:
Can it happen that there is no failure? Can we prevent that? It would be good to make sure that these exceptions always carry a meaningful message.

Reply:
What is the rationale for using the first failure as the message? It could also be "<N> errors were encountered during validation". We certainly don't want to use the word "Exception" because that is just the Java implementation of an error.

Reply:
To prevent empty list of failures, we throw the `ValidationException` from stage configurer only when there are any failures added to the `ValidationException`. However, since `ValidationException` is part of the api and plugins can still create an instance of the exception with an empty list and throw it. Thats why I have added the check for empty failures. 
Fixed the error message.